Sources/PenguinStructures/Graphs/AdjacencyList.swift

```diff
@@ -101,42 +118,60 @@ extension AdjacencyList: MutableGraph {
 		edgesArray[edge.source.index].remove(at: edge.offset)
 	}
 
+    /// Removes all edges that satisfy `predicate`.
 	public mutating func removeEdges(_ predicate: (EdgeId) throws -> Bool) rethrows {
 		for sourceId in 0..<edgesArray.count {
 			try removeEdges(from: VertexId(IdType(sourceId)), predicate)
 		}
 	}
 
-	public mutating func removeEdges(from node: VertexId, _ predicate: (EdgeId) throws -> Bool) rethrows {
+    /// Remove all out edges from `vertex` that satisfy the given predicate.
+    ///
+    /// - Complexity: O(|E|)
+	public mutating func removeEdges(from vertex: VertexId, _ predicate: (EdgeId) throws -> Bool) rethrows {
 		var shouldRemove = Set<IdType>()
-		let vertexEdges = edgesArray[node.index]
+		let vertexEdges = edgesArray[vertex.index]
 		for (i, dest) in vertexEdges.enumerated() {
-			let edgeId = EdgeId(source: node, destination: VertexId(dest), offset: i)
+			let edgeId = EdgeId(source: vertex, destination: VertexId(dest), offset: i)
 			if try predicate(edgeId) {
 				shouldRemove.insert(dest)
 			}
 		}
-		edgesArray[node.index].removeAll { shouldRemove.contains($0) }
+		edgesArray[vertex.index].removeAll { shouldRemove.contains($0) }
 	}
 
+    /// Adds a new vertex to the graph, and returns its identifier.
+    ///
+    /// - Complexity: O(1) (amortized)
 	public mutating func addVertex() -> VertexId {
 		let c = edgesArray.count
 		edgesArray.append([])
 		return VertexId(IdType(c))
 	}
 
+    /// Removes all edges from `vertex`.
+    ///
+    /// - Complexity: O(|E|)
 	public mutating func clear(vertex: VertexId) {
 		edgesArray[vertex.index].removeAll()
 	}
 
+    /// Removes `vertex` from the graph.
+    ///
+    /// - Precondition: `vertex` is a valid `VertexId` for `self`.
+    /// - Complexity: O(|E| + |V|)
 	public mutating func remove(vertex: VertexId) {
 		fatalError("Unimplemented!")
 	}
 }
 
 extension AdjacencyList: VertexListGraph {
+    /// The total number of verticies in the graph.
+    ///
+    /// - Complexity: O(1)
```

Comment:
You shouldn't have to say this for a property.  IMO the right standard is, if you have a property that's not O(1), *that* needs to be documented.